File: arduinobot_ws/src/arduinobot_test/scripts/simple_service.py
#!/usr/bin/env python3
from arduinobot_test.srv import AddTwoInts, AddTwoIntsResponse
import rospy


def add_two_ints(req):
    # Function that is called each time the service receives a request
    # The request message is passed as input to this function
    rospy.loginfo('Ready to sum %d and %d', req.a, req.b)
    return AddTwoIntsResponse(req.a + req.b)

if __name__ == "__main__":
    # Inizialize a ROS node called simple_service
    rospy.init_node('simple_service')

    # Initialize a service named add_two_ints that uses the AddTwoInts
    # for the Request/Response communication interface
    # Define the function that is called each time the service receives a request
    service = rospy.Service('add_two_ints', AddTwoInts, add_two_ints)

    rospy.loginfo("Ready to add two ints.")

    # keeps the node up and running
    rospy.spin()

File: arduinobot_ws/src/arduinobot_test/src/simple_service.cpp
#include "ros/ros.h"
#include "arduinobot_test/AddTwoInts.h"


bool add(arduinobot_test::AddTwoInts::Request  &req,
         arduinobot_test::AddTwoInts::Response &res)
{
    // Function that is called each time the service receives a request
    // The request message is passed as input to this function
    ROS_INFO("Ready to sum %d and %d", (int)req.a, (int)req.b);
    res.sum = req.a + req.b;
    return true;
}

int main(int argc, char **argv)
{
    // Inizialize a ROS node called simple_service
    ros::init(argc, argv, "simple_service");
    ros::NodeHandle n;

    // Initialize a service named add_two_ints that uses the AddTwoInts
    // for the Request/Response communication interface
    // Define the function that is called each time the service receives a request
    ros::ServiceServer service = n.advertiseService("add_two_ints", add);
    ROS_INFO("The service is ready to add two ints.");

    // keeps the node up and running
    ros::spin();
    return 0;
}